From f474ba848a44ab8df111e74e945aa60a05d7194f Mon Sep 17 00:00:00 2001
From: Axel Castaneda Gonzalez <x0055901@ti.com>
Date: Wed, 23 Mar 2011 20:29:53 -0500
Subject: ASoC: HDMI: Remove HDMI IP module support

Remove deprecated HDMI IP module support since its
design is not functional anymore.

Change-Id: I8bba06a0f8c0cfc681008cacb02e3dd34b478ec8
Signed-off-by: Axel Castaneda Gonzalez <x0055901@ti.com>
---
 sound/soc/omap/omap-hdmi.c |   78 +------------------------------------------
 1 files changed, 2 insertions(+), 76 deletions(-)

Index: kernel/sound/soc/omap/omap-hdmi.c
===================================================================
--- kernel.orig/sound/soc/omap/omap-hdmi.c	2011-04-07 11:56:04.000000000 +0900
+++ kernel/sound/soc/omap/omap-hdmi.c	2011-04-07 12:11:22.000000000 +0900
@@ -32,6 +32,7 @@
 
 #include <plat/control.h>
 #include <plat/dma.h>
+#include <plat/hdmi_lib.h>
 #include <plat/omap-pm.h>
 #include "omap-pcm.h"
 #include "omap-hdmi.h"
@@ -47,7 +48,7 @@
 /* Currently, we support only 16b samples at HDMI */
 #define OMAP_HDMI_FORMATS (SNDRV_PCM_FMTBIT_S16_LE)
 
-#ifdef CONFIG_HDMI_NO_IP_MODULE
+
 #include <plat/hdmi_lib.h>
 
 struct omap_hdmi_data {
@@ -58,9 +59,6 @@
 };
 
 struct omap_hdmi_data hdmi_data;
-#else
-struct hdmi_ip_driver hdmi_audio_core;
-#endif
 
 static struct omap_pcm_dma_data omap_hdmi_dai_dma_params = {
 	.name = "HDMI playback",
@@ -69,7 +67,6 @@
 	.sync_mode = OMAP_DMA_SYNC_PACKET,
 };
 
-#ifdef CONFIG_HDMI_NO_IP_MODULE
 static void hdmi_hpd_notifier(int state, void *data)
 {
 	struct omap_hdmi_data *hdmi_data = data;
@@ -115,13 +112,11 @@
 		break;
 	}
 }
-#endif
 
 static int omap_hdmi_dai_startup(struct snd_pcm_substream *substream,
 				  struct snd_soc_dai *dai)
 {
 	int err = 0;
-#ifdef CONFIG_HDMI_NO_IP_MODULE
 	if (!hdmi_data.active) {
 		printk(KERN_ERR "hdmi device not available\n");
 		return -ENODEV;
@@ -136,12 +131,6 @@
 	hdmi_set_audio_power(1);
 	hdmi_data.substream = substream;
 	err = hdmi_w1_wrapper_enable(HDMI_WP);
-#else
-	if (hdmi_audio_core.module_loaded)
-		err = hdmi_audio_core.wrapper_enable(HDMI_WP);
-	else
-		printk(KERN_WARNING "Warning: hdmi_core.ko is not enabled");
-#endif
 	return err;
 }
 
@@ -149,7 +138,6 @@
 				    struct snd_soc_dai *dai)
 {
 	int err = 0;
-#ifdef CONFIG_HDMI_NO_IP_MODULE
 	hdmi_w1_wrapper_disable(HDMI_WP);
 	hdmi_data.substream = NULL;
 
@@ -169,13 +157,6 @@
 		printk(KERN_WARNING
 			"unable to remove SDMA constraint %d\n", err);
 #endif
-#else
-	if (hdmi_audio_core.module_loaded)
-		hdmi_audio_core.wrapper_disable(HDMI_WP);
-	else
-		printk(KERN_WARNING "Warning: hdmi_core.ko is not enabled");
-#endif
-	return;
 }
 
 static int omap_hdmi_dai_trigger(struct snd_pcm_substream *substream, int cmd,
@@ -187,35 +168,19 @@
 	case SNDRV_PCM_TRIGGER_START:
 	case SNDRV_PCM_TRIGGER_RESUME:
 	case SNDRV_PCM_TRIGGER_PAUSE_RELEASE:
-#ifdef CONFIG_HDMI_NO_IP_MODULE
 		if (cmd == SNDRV_PCM_TRIGGER_RESUME)
 			hdmi_set_audio_power(1);
 		err = hdmi_w1_start_audio_transfer(HDMI_WP);
-#else
-		if (hdmi_audio_core.module_loaded)
-			err = hdmi_audio_core.start_audio(HDMI_WP);
-		else
-			printk(KERN_WARNING "Warning: hdmi_core.ko is "
-							"not enabled");
-#endif
 		break;
 
 	case SNDRV_PCM_TRIGGER_STOP:
 	case SNDRV_PCM_TRIGGER_SUSPEND:
 	case SNDRV_PCM_TRIGGER_PAUSE_PUSH:
-#ifdef CONFIG_HDMI_NO_IP_MODULE
 		err = hdmi_w1_stop_audio_transfer(HDMI_WP);
 		if (err)
 			return err;
 		if (cmd == SNDRV_PCM_TRIGGER_SUSPEND)
 			hdmi_set_audio_power(0);
-#else
-		if (hdmi_audio_core.module_loaded)
-			err = hdmi_audio_core.stop_audio(HDMI_WP);
-		else
-			printk(KERN_WARNING "Warning: hdmi_core.ko is "
-							"not enabled");
-#endif
 		break;
 	default:
 		err = -EINVAL;
@@ -317,43 +282,6 @@
 }
 module_exit(hdmi_dai_exit);
 
-#ifndef CONFIG_HDMI_NO_IP_MODULE
-
-/* stub */
-int audio_stub_lib_init(void)
-{
-	printk(KERN_WARNING "ERR: please install HDMI IP kernel module\n");
-	return -1;
-}
-void audio_stub_lib_exit(void)
-{
-	printk(KERN_WARNING "HDMI module does not exist!\n");
-}
-
-#define EXPORT_SYMTAB
-
-/* HDMI panel driver */
-void hdmi_audio_core_stub_init(void)
-{
-	hdmi_audio_core.stop_video = NULL;
-	hdmi_audio_core.start_video = NULL;
-	hdmi_audio_core.wrapper_enable = NULL;
-	hdmi_audio_core.wrapper_disable = NULL;
-	hdmi_audio_core.stop_audio = NULL;
-	hdmi_audio_core.start_audio = NULL;
-	hdmi_audio_core.config_video = NULL;
-	hdmi_audio_core.set_wait_pll = NULL;
-	hdmi_audio_core.set_wait_pwr = NULL;
-	hdmi_audio_core.set_wait_srst = NULL;
-	hdmi_audio_core.read_edid = NULL;
-	hdmi_audio_core.ip_init = audio_stub_lib_init;
-	hdmi_audio_core.ip_exit = audio_stub_lib_exit;
-	hdmi_audio_core.module_loaded = 0;
-}
-
-#endif
-
-
 MODULE_AUTHOR("Jorge Candelaria <x0107209@ti.com");
 MODULE_DESCRIPTION("OMAP HDMI SoC Interface");
 MODULE_LICENSE("GPL");
